src/groupByKey.ts

```diff
@@ -2,6 +2,27 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<I>(items: I[], key: keyof I): object {
+  let groupKeys = items.map((item: I) => item[key]);
+
+  groupKeys
+    = groupKeys
+      .filter((value, index: number) => groupKeys.indexOf(value) === index);
+
+  const maps
+    = groupKeys
+      .map((groupKey) => {
+        const map: GroupsMap<I>
+          = {
+            [`${groupKey}`]: items.filter((item : I) => item[key] === groupKey),
+          };
+
+        return map;
```

Comment:
maybe we can just return map without creating const?

Reply:
++++